tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -1258,6 +1258,44 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the TL-WA850RE v4 */
+	{
+		.id     = "TLWA850REV4",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:55530000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:45550000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:4B520000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:42520000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:4A500000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:41550000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:41520000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:54570000}\n"
+			"{product_name:TL-WA850RE,product_ver:4.0.0,special_id:45530000}\n",
+		.support_trail = '\x00',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"firmware", 0x20000, 0x3a2000},
```

Comment:
That's not matching your DTS.

Reply:
So, what's inside these 0x2000?

Reply:
0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF............
You're the one who complained about me leaving the gap there.

Reply:
I could leave the gap in the dts too ... it would make the area even more unwritable-to by the MTD system (since not aligned to erase blocks) and leave a weird gap since i cant move the firmware area end in the dts forward because that'd make the firmware area unwritable to...

Reply:
I never complain, I trigger discussions about questions to be solved. ;-)

I remember a similar case where extending the firmware partition caused problems.

Reply:
if I have to choose between gap and inconsistency between safeloader and DTS, I choose the gap